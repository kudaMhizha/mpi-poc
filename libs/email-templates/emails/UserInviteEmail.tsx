import React from 'react';
import {
  Body,
  Container,
  Column,
  Head,
  Heading,
  Html,
  Link,
  Preview,
  Row,
  Section,
  Text,
} from '@react-email/components';

type InviteEmailProps = {
  code?: string;
  title?: string;
  message?: string;
  helperText?: string;
  appLink?: string;
};

export function UserInviteEmail({
  code = 'A77IiY1c!k!c',
  title = 'Join Company Name',
  message = `Your email address (email@email) and your temporary code is below - enter it and sign in.`,
  helperText = '',
  appLink,
}: Readonly<InviteEmailProps>): JSX.Element {
  return (
    <Html>
      <Head />
      <Preview>Confirm your email address</Preview>
      <Body style={main}>
        <Container style={container}>
          <Heading style={h1}>{title}</Heading>
          <Text style={heroText}>{message}</Text>
          <Section style={codeBox}>
            <Text style={confirmationCodeText}>{code}</Text>
          </Section>
          <Text style={text}>{helperText}</Text>
          <Section
            style={{
              marginTop: '20px',
              marginBottom: '65px',
              display: 'flex',
              alignContent: 'center',
              justifyContent: 'center',
            }}
          >
            <Row>
              <Column>
                <Link
                  style={button}
                  href={`https://${appLink}`}
                  target="_blank"
                >
                  Sign Up
                </Link>
              </Column>
            </Row>
          </Section>
          <Text style={text}>
            If you didn't request this email, there's nothing to worry about -
            you can safely ignore it.
          </Text>
        </Container>
      </Body>
    </Html>
  );
}

export default UserInviteEmail;

const button = {
  backgroundColor: '#f3f3f5',
  fontSize: '17px',
  lineHeight: '17px',
  padding: '13px 17px',
  borderRadius: '24px',
  maxWidth: '120px',
  color: '#2E4A49',
};

const main = {
  backgroundColor: '#ffffff',
  margin: '0 auto',
  fontFamily:
    "-apple-system, BlinkMacSystemFont, 'Segoe UI', 'Roboto', 'Oxygen', 'Ubuntu', 'Cantarell', 'Fira Sans', 'Droid Sans', 'Helvetica Neue', sans-serif",
};

const container = {
  maxWidth: '600px',
  margin: '0 auto',
};

const h1 = {
  color: '#1d1c1d',
  fontSize: '36px',
  fontWeight: 700,
  margin: '30px 0',
  padding: '0',
  lineHeight: '42px',
};

const heroText = {
  fontSize: '20px',
  lineHeight: '28px',
  marginBottom: '30px',
};

const codeBox = {
  background: 'rgb(245, 244, 245)',
  borderRadius: '4px',
  marginRight: '50px',
  marginBottom: '30px',
  padding: '43px 23px',
};

const confirmationCodeText = {
  fontSize: '30px',
  textAlign: 'center' as const,
  verticalAlign: 'middle',
};

const text = {
  color: '#000',
  fontSize: '14px',
  lineHeight: '24px',
};
