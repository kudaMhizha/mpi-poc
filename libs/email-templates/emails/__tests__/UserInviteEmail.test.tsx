import renderer from 'react-test-renderer';
import UserInviteEmail from '../UserInviteEmail';
it('renders correctly', () => {
  const tree = renderer.create(<UserInviteEmail />).toJSON();
  expect(tree).toMatchSnapshot();
});
