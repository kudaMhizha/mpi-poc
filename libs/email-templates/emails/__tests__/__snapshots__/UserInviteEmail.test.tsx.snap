// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<html
  dir="ltr"
  lang="en"
>
  <head>
    <meta
      content="text/html; charset=UTF-8"
      httpEquiv="Content-Type"
    />
  </head>
  <div
    style={
      Object {
        "display": "none",
        "lineHeight": "1px",
        "maxHeight": 0,
        "maxWidth": 0,
        "opacity": 0,
        "overflow": "hidden",
      }
    }
  >
    Confirm your email address
    <div>
       ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿
    </div>
  </div>
  <body
    style={
      Object {
        "backgroundColor": "#ffffff",
        "fontFamily": "-apple-system, BlinkMacSystemFont, 'Segoe UI', 'Roboto', 'Oxygen', 'Ubuntu', 'Cantarell', 'Fira Sans', 'Droid Sans', 'Helvetica Neue', sans-serif",
        "margin": "0 auto",
      }
    }
  >
    <table
      align="center"
      border={0}
      cellPadding="0"
      cellSpacing="0"
      role="presentation"
      style={
        Object {
          "margin": "0 auto",
          "maxWidth": "600px",
        }
      }
      width="100%"
    >
      <tbody>
        <tr
          style={
            Object {
              "width": "100%",
            }
          }
        >
          <td>
            <h1
              style={
                Object {
                  "color": "#1d1c1d",
                  "fontSize": "36px",
                  "fontWeight": 700,
                  "lineHeight": "42px",
                  "margin": "30px 0",
                  "padding": "0",
                }
              }
            >
              Join Company Name
            </h1>
            <p
              style={
                Object {
                  "fontSize": "20px",
                  "lineHeight": "28px",
                  "margin": "16px 0",
                  "marginBottom": "30px",
                }
              }
            >
              Your email address (email@email) and your temporary code is below - enter it and sign in.
            </p>
            <table
              align="center"
              border={0}
              cellPadding="0"
              cellSpacing="0"
              role="presentation"
              style={
                Object {
                  "background": "rgb(245, 244, 245)",
                  "borderRadius": "4px",
                  "marginBottom": "30px",
                  "marginRight": "50px",
                  "padding": "43px 23px",
                }
              }
              width="100%"
            >
              <tbody>
                <tr>
                  <td>
                    <p
                      style={
                        Object {
                          "fontSize": "30px",
                          "lineHeight": "24px",
                          "margin": "16px 0",
                          "textAlign": "center",
                          "verticalAlign": "middle",
                        }
                      }
                    >
                      A77IiY1c!k!c
                    </p>
                  </td>
                </tr>
              </tbody>
            </table>
            <p
              style={
                Object {
                  "color": "#000",
                  "fontSize": "14px",
                  "lineHeight": "24px",
                  "margin": "16px 0",
                }
              }
            />
            <table
              align="center"
              border={0}
              cellPadding="0"
              cellSpacing="0"
              role="presentation"
              style={
                Object {
                  "alignContent": "center",
                  "display": "flex",
                  "justifyContent": "center",
                  "marginBottom": "65px",
                  "marginTop": "20px",
                }
              }
              width="100%"
            >
              <tbody>
                <tr>
                  <td>
                    <table
                      align="center"
                      border={0}
                      cellPadding="0"
                      cellSpacing="0"
                      role="presentation"
                      width="100%"
                    >
                      <tbody
                        style={
                          Object {
                            "width": "100%",
                          }
                        }
                      >
                        <tr
                          style={
                            Object {
                              "width": "100%",
                            }
                          }
                        >
                          <td
                            data-id="__react-email-column"
                          >
                            <a
                              href="https://undefined"
                              style={
                                Object {
                                  "backgroundColor": "#f3f3f5",
                                  "borderRadius": "24px",
                                  "color": "#2E4A49",
                                  "fontSize": "17px",
                                  "lineHeight": "17px",
                                  "maxWidth": "120px",
                                  "padding": "13px 17px",
                                  "textDecoration": "none",
                                }
                              }
                              target="_blank"
                            >
                              Sign Up
                            </a>
                          </td>
                        </tr>
                      </tbody>
                    </table>
                  </td>
                </tr>
              </tbody>
            </table>
            <p
              style={
                Object {
                  "color": "#000",
                  "fontSize": "14px",
                  "lineHeight": "24px",
                  "margin": "16px 0",
                }
              }
            >
              If you didn't request this email, there's nothing to worry about - you can safely ignore it.
            </p>
          </td>
        </tr>
      </tbody>
    </table>
  </body>
</html>
`;
